### DISCUSSION TITLE
RuntimeError: applying transform <monai.transforms.io.dictionary.LoadImaged object at 0x7f65ffee7f70>  ,When I setting 2D classification Dataset.

### DISCUSSION THREAD

**Discussion by danny0628** [Project-MONAI/MONAI#3693] [Q&A] [answered]
I am preparing an MNIST classification dataset, and the image is ready in png format, and the json file is prepared as follows, but an error occurs. (I used Monai in Nvidia Clara SDK)

> this is datset.json,

{ 
    "label_format": [1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],
    "training":[
	{
		"image":"1.png",
		"label":[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
	},

> this is important part of train_config.json,

  "train": {
    "model": {
      "name": "DenseNet",
      "args": {
        "spatial_dims": 2,
        "in_channels": 1,
        "out_channels": 2
      }
    },
    "loss": {
      "name": "CrossEntropyLoss"
    },
    "optimizer": {
      "name": "Adam"
    },
    "pre_transforms": [
      {
        "name": "LoadImaged",
        "args": {
          "keys": ["image", "label"]
        },
        "reader":"PILReader",
        "image_only":"True"
      },
....
}

> PYTHONPATH is :/opt/nvidia/medical:/opt/nvidia:/opt/nvidia:/claraDevDay/GettingStarted/commands/../custom
MMAR_ROOT set to /claraDevDay/GettingStarted/commands/..
2022-01-21 08:20:27,581 - torch.distributed.nn.jit.instantiator - INFO - Created a temporary directory at /tmp/tmpeju4tw31
2022-01-21 08:20:27,581 - torch.distributed.nn.jit.instantiator - INFO - Writing /tmp/tmpeju4tw31/_remote_module_non_sriptable.py
Loading dataset:   0%|                                    | 0/4 [00:00<?, ?it/s]input data information of the runtime error transform check error porint!!!!!:
input data information of the runtime error transform check error porint!!!!!:
input data information of the runtime error transform check error porint!!!!!:
2022-01-21 08:20:31,132 - DataStats - INFO - input data information of the runtime error transform check error porint!!!!!:
input data information of the runtime error transform check error porint!!!!!:
image statistics:
Type: <class 'str'>
Value: /claraDevDay/Data/mnist/images/21.png
2022-01-21 08:20:31,132 - DataStats - INFO - input data information of the runtime error transform check error porint!!!!!:
image statistics:
Type: <class 'str'>
Value: /claraDevDay/Data/mnist/images/21.png
image statistics:
Type: <class 'str'>
Value: /claraDevDay/Data/mnist/images/1.png
2022-01-21 08:20:31,133 - DataStats - INFO - image statistics:
Type: <class 'str'>
Value: /claraDevDay/Data/mnist/images/21.png
image statistics:
Type: <class 'str'>
Value: /claraDevDay/Data/mnist/images/1.png
label statistics:
Type: <class 'list'>
Value: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
2022-01-21 08:20:31,133 - DataStats - INFO - image statistics:
Type: <class 'str'>
Value: /claraDevDay/Data/mnist/images/1.png
label statistics:
Type: <class 'list'>
Value: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
label statistics:
Type: <class 'list'>
Value: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
2022-01-21 08:20:31,133 - DataStats - INFO - label statistics:
Type: <class 'list'>
Value: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
label statistics:
Type: <class 'list'>
Value: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
2022-01-21 08:20:31,133 - DataStats - INFO - label statistics:
Type: <class 'list'>
Value: [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
Loading dataset:   0%|                                    | 0/4 [00:00<?, ?it/s]
Error processing config /claraDevDay/GettingStarted/commands/../config/config_train.json: applying transform <monai.transforms.io.dictionary.LoadImaged object at 0x7f65ffee7f70>
> 

Please help control Datset Format :)

**Comment by Nic-Ma** [comment, marked as answer]
Hi @danny0628 ,

Thanks for your interest in MONAI and Clara Train SDK.
I feel your transform and network setting is not correct:
```
"name": "LoadImaged",
"args": {
"keys": ["image", "label"]    --> your label is not image, should not in `LoadImaged`
},
```
And I saw every label has 15 classes: `"label":[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0]`, but your DenseNet output 2 channels?

Thanks.

**Reply by danny0628** [reply]
Thanks for reply!!! :)
I change 15 classes -> 2 classes
and,
Do I change ["image","label"]   ---> ["image"] ? or Do I have to remove all of them?

**Reply by Nic-Ma** [reply]
Please change `change ["image","label"] ---> ["image"]` and try.

Thanks.
